### ISSUE TITLE
New oauth2 versions can't authenticate

### ISSUE THREAD

**Issue by cavis** [PRX/pmp#9]
For some reason, oauth2 version 1.4.2 can't authenticate with the PMP using client credentials.  It gets an unauthorized.  Something appears to have broken between 1.0.0 and 1.4.2.

Should try to isolate/fix that, or at least lock the gemspec to avoid bad versions.

**Comment by andreibondarev** [comment]
@cavis My build fully passes with `gem 'oauth2', '1.2.0'`. Must be a version between 1.2.0 and 1.4.2.

**Comment by kookster** [comment]
yeah, when I test locally, something weird - doesn't look like it is setting the `Authorization` header in the expected manner in the test failure output (i.e. the test expects an Authorization header, but it isn't getting set):
```
stub_request(:post, "https://api.pmp.io/auth/access_token").
  with(
    body: {"client_id"=>"thisisnota-real-client-id-soverysorry", "client_secret"=>"thisisnotarealsecreteither", "grant_type"=>"client_credentials"},
    headers: {
	  'Accept'=>'application/json',
	  'Content-Type'=>'application/x-www-form-urlencoded',
	  'Host'=>'api.pmp.io:443',
	  'User-Agent'=>'PMP Ruby Gem 0.5.6'
    }).
  to_return(status: 200, body: "", headers: {})

registered request stubs:

stub_request(:post, "https://api.pmp.io/auth/access_token").
  with(
    body: {"grant_type"=>"client_credentials"},
    headers: {
	  'Accept'=>'application/json',
	  'Authorization'=>'Basic dGhpc2lzbm90YS1yZWFsLWNsaWVudC1pZC1zb3Zlcnlzb3JyeTp0aGlzaXNub3RhcmVhbHNlY3JldGVpdGhlcg==',
	  'Content-Type'=>'application/x-www-form-urlencoded'
    })
```

**Comment by andreibondarev** [comment]
May want to release `0.5.7` locking the `oauth2` version before this fix is in place.

**Comment by kookster** [comment]
Yeah, I am about to make a PR with the gem locked, but wanted to take a brief look at the issue.

I'm suspicious of this commit:

https://github.com/oauth-xx/oauth2/commit/14ab0e000a84f1012d3000ee2b39bb74fb172b36#diff-bc9331245a41740ee0145e2d5640ed07R193-R199

which creates a condition where the Authorization header won't be set if there is a request body...

**Comment by kookster** [comment]
it also is the right timing, it was released with 1.3.0, which has this issue for our tests

**Comment by kookster** [comment]
`0.5.7` released with the spec pinned to oauth2 < 1.3.0
